docs/integrations/httpx.md

```diff
@@ -63,6 +63,152 @@ Let's see a minimal example below. You can run it with `python main.py`:
     asyncio.run(main())
     ```
 
+## Hooks
+
+The `logfire.instrument_httpx()` also accepts the following hooks:
+
+- `request_hook`: called before sending a request.
+- `response_hook`: called after receiving a response.
+- `async_request_hook`: called before sending an async request.
+- `async_response_hook`: called after receiving an async response.
+
+You can use these hooks to add custom attributes that are not automatically captured by **Logfire**.
+
+### (Async) Request Hook
+
+The `request_hook` and `async_request_hook` hooks are called before sending a request. They receive two arguments:
+
+```py title="main.py" hl_lines="24-27"
+import asyncio
+from collections import defaultdict
+
+import httpx
+from opentelemetry.instrumentation.httpx import RequestInfo
+from opentelemetry.trace import Span
+
+import logfire
+
+
+def set_request_headers(span: Span, request: RequestInfo):
+    method, url, headers, stream, extensions = request
+    attributes: defaultdict[str, str] = defaultdict(list)
+    for key, value in headers.items():
+        key = key.lower()
+        attributes[f'http.request.header.{key}'].append(value)
```

Comment:
I started working on capturing headers. Currently that code is stashed. Once pushed this will become a bad example.

Reply:
Can you suggest a useful example? I couldn't think of any...